browser/ui/views/omnibox/brave_omnibox_result_view.cc

```diff
@@ -39,16 +41,18 @@
 BraveOmniboxResultView::~BraveOmniboxResultView() = default;
 
 void BraveOmniboxResultView::ResetChildren() {
-  // Reset children visibility. Their visibility could be configured later
-  // based on |match_| and the current input.
-  // NOTE: The first child in the result box is supposed to be the
-  // `suggestion_container_`, which used to be stored as a data member.
-  children().front()->SetVisible(true);
-  button_row_->SetVisible(true);
   if (brave_search_promotion_view_) {
     RemoveChildViewT(brave_search_promotion_view_);
     brave_search_promotion_view_ = nullptr;
   }
+
+  // Reset children visibility. Their visibility could be configured later
+  // based on |match_| and the current input.
+  // Reset upstream's layout manager.
+  SetLayoutManager(std::make_unique<views::FillLayout>());
```

Comment:
Out of curiosity, is this needed because FlexLayout might have been set?

Reply:
If not set, original result view layout is broken. OmniboxResultView uses `FillLayout`.
![image](https://github.com/brave/brave-core/assets/6786187/ca4e29dd-f884-45ab-8265-edfcf9750ae6)